src/tea_app.ml

```diff
@@ -8,416 +8,259 @@
    Tea.AppEx or so as a simple wrapper? *)
 
 
-(* module type Program = sig
```

Comment:
I tried to get modules to fit Elm's style but it does not match it's API and makes porting more difficult.  I already have a module style planned for the Elm-API-breaking style though, but the base TEA must follow Elm's style.